File: Modulo3/persistencia/src/main/java/com/codigo/persistencia/service/CategoriaService.java
package com.codigo.persistencia.service;


import com.codigo.persistencia.dao.CategoriaRepository;
import com.codigo.persistencia.entity.Categoria;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
    @AllArgsConstructor
    public class CategoriaService {
        private final CategoriaRepository categoriaRepository;

        @Transactional
        public Categoria registrarCategoria(Categoria categoria)
        {
            return categoriaRepository.save(categoria);

        }
}

File: Modulo3/persistencia/src/main/java/com/codigo/persistencia/service/CategoriaService.kt
package com.codigo.persistencia.service


import lombok.AllArgsConstructor
import org.springframework.stereotype.Service

import org.springframework.transaction.annotation.Transactional

@Service
@AllArgsConstructor
class CategoriaService {
    private final CategoriaRepository categoriaRepository;

    @Transactional
    public Categoria registrarCategoria(Categoria categoria)
    {
        return categoriaRepository.save(categoria);

    }

}